anda/apps/openasar-canary/chkupdate.py

```diff
@@ -0,0 +1,23 @@
+import requests
+import re
+import os
+
+
+NAME = "openasar-canary"
+REPO = "GooseMod/OpenAsar"
+REGEX_SHA = r"%define commit (.+)"
+SPEC = f"{NAME}.spec"
+LINK = f"https://api.github.com/repos/{REPO}/commits/HEAD"
+
+token = os.getenv("GITHUB_TOKEN")
+sha = requests.get(LINK, headers={"Authorization": f"Bearer {token}"}).json()["sha"]
+f = open(SPEC, "r")
+
+matches = re.findall(REGEX_SHA, txt := f.read())
+if not len(matches):
+    exit(f"{NAME}: Failed to match regex!")
+cur = matches[0]
+
+if sha == cur:
+    exit(f"{NAME}: Up to date!")
+print(f"{NAME}: {cur} -> {sha}")
```

Comment:
so this file writes nothing?

Reply:
yep, the version is already determined with [%{autogitdate}](https://github.com/terrapkg/packages/blob/44cf5b5a477d3100daee78cca28c344a685fe750/anda/apps/openasar-canary/openasar-canary.spec#L4) and the [source](https://github.com/terrapkg/packages/blob/44cf5b5a477d3100daee78cca28c344a685fe750/anda/apps/openasar-canary/openasar-canary.spec#L10) just points to the continuously updated nightly release upstream. the goal of this script was to just trigger a rebuild on each new commit (which should signify a new nightly release), which is why i started defining the commit in the spec file

Reply:
I don't think builds will trigger with no files changed in the dir…

Reply:
🤦🏻 i see what you mean now. i'll push another commit now

...maybe i shouldn't be doing this at 4am